bionic/cache.py

```diff
@@ -746,16 +747,14 @@ def _validated_object_name_from_url(self, url):
         return object_name
 
     def _bucket_and_object_names_from_url(self, url):
-        if not url.startswith(self._GS_URL_PREFIX):
-            raise ValueError(f'url must start with "{self._GS_URL_PREFIX}"')
-        url_parts = url[len(self._GS_URL_PREFIX):].split('/', 1)
-        if len(url_parts) == 1:
-            bucket_name, = url_parts
-            object_prefix = ''
-        else:
-            bucket_name, object_prefix = url_parts
-
-        return bucket_name, object_prefix
+        if not is_gcs_url(url):
+            raise ValueError(f'url must have schema "{GCS_SCHEME}"')
+        result = urlparse(url)
+        object_name = result.path
```

Comment:
super nit: this isn't an `object_name`, it's either a `result_path` or an `object_name_with_leading_slash`. :)